src/transformers/modeling_utils.py

```diff
@@ -781,15 +781,21 @@ def generate(
         cur_len = input_ids.shape[1]
         vocab_size = self.config.vocab_size
 
-        if num_return_sequences != 1 and do_sample:
-            # Expand input to num return sequences
-            input_ids = input_ids.unsqueeze(1).expand(batch_size, num_return_sequences, cur_len)
-            input_ids = input_ids.contiguous().view(
-                batch_size * num_return_sequences, cur_len
-            )  # shape: (batch_size * num_return_sequences, cur_len)
+        # set effective batch size and effective batch multiplier according to do_sample
```

Comment:
Here I included the lines 935 - 938 and did some renaming. 
The reason for this are:

- Consistency and readability: input_ids are expanded only once in the beginning allowing the functions
_no_beam_search_generation() and _beam_search_generation() to have a similar structure in the beginning and avoids code duplication
- Is more logical: input_ids are only expanded if either num_return_sequences > 1 or num_beams > 1 regardless of other parameters. 
@thomwolf @LysandreJik